ftdc/ftdc.go

```diff
@@ -129,14 +129,16 @@ func (ftdc *FTDC) Add(name string, statser Statser) {
 
 	for _, statser := range ftdc.statsers {
 		if statser.name == name {
-			ftdc.logger.Warnw("Trying to add conflicting ftdc section", "name", name)
+			ftdc.logger.Warnw("Trying to add conflicting ftdc section", "name", name,
+				"generationId", ftdc.inputGenerationID)
```

Comment:
Figured if I was going to do this optimistic concurrency control stuff, I'd add the concurrency mechanism values to the relevant log output.

Reply:
Could you briefly explain the difference between `inputGenerationID` and `outputGenerationID`?

Reply:
We have to write a new schema when the set of `Statser`s change. We use "generation id"s to track when the set of `Statser`s change.

`inputGenerationID` increments when `Statser`s are added or removed. When we "create a datum", that has exactly the statsers described by the `inputGenerationID`.

`outputGenerationID` represents the set of `Statser`s/fields last written.

When `inputGenerationID == outputGenerationID` (the steady state), we don't need to write a new schema. We can just write datum values.

But when `inputGenerationID > outputGenerationID`, we have to write a new schema. Then we can update the `outputGenerationID`.

And above is supposed to be a small simplification. We don't compare input to output directly. These are numbers communicated via the datums.

Reply:
Super helpful; thanks. Think I get it now.